File: leetcode/easy/count_good_triplets.go
package main

import (
	"fmt"
)

func abs(num int) int {
	if num < 0 {
		return -num
	}
	return num
}

func countGoodTriplets(arr []int, a int, b int, c int) int {
	result := 0
	length := len(arr)

	for i := 0; i < length; i++ {
		for j := i + 1; j < length; j++ {
			for k := j + 1; k < length; k++ {
				if abs(arr[i]-arr[j]) <= a && abs(arr[j]-arr[k]) <= b && abs(arr[i]-arr[k]) <= c {
					result += 1
				}
			}
		}
	}

	return result
}

func main() {
	fmt.Println(countGoodTriplets([]int{3, 0, 1, 1, 9, 7}, 7, 2, 3))
	fmt.Println(countGoodTriplets([]int{1, 1, 2, 2, 3}, 0, 0, 1))
}

File: leetcode/easy/count-good-triplets.cpp
#include <iostream>
#include <vector>

using namespace std;

int countGoodTriplets(vector<int>& arr, int a, int b, int c) {
    int count = 0;
    int size = arr.size();

    for (int i = 0; i < size; i++) {
        for (int j = i + 1; j < size; j++) {
        for (int k = j + 1; k < size; k++) {
            if (abs(arr[i] - arr[j]) <= a && abs(arr[j] - arr[k]) <= b &&
                abs(arr[i] - arr[k]) <= c) {
            count++;
            }
        }
        }
    }
    return count;
}

int main() {
  vector<int> nums{3, 0, 1, 1, 9, 7};
  int result = countGoodTriplets(nums, 7, 2, 3);

  cout << result;
}